sys/bloom/bloom.c

```diff
@@ -35,7 +35,7 @@ bloom_t *bloom_new(size_t size, size_t num_hashes, ...)
     }
 
     /* Allocate Bloom array */
-    if (!(bloom->a = calloc(ROUND(size), sizeof(char)))) {
+    if (!(bloom->a = calloc(1, ROUND(size)))) {
```

Comment:
The type of bloom->a is uint8_t* , not char*, I guess that's why the
warning appeared.
On Aug 24, 2015 10:44 AM, "René Kijewski" <notifications@github.com> wrote:

> In sys/bloom/bloom.c
> <https://github.com/RIOT-OS/RIOT/pull/3699#discussion_r37730074>:
>
> > @@ -35,7 +35,7 @@ bloom_t *bloom_new(size_t size, size_t num_hashes, ...)
> >      }
> >
> >      /* Allocate Bloom array */
> > -    if (!(bloom->a = calloc(ROUND(size), sizeof(char)))) {
> > +    if (!(bloom->a = calloc(1, ROUND(size)))) {
>
> The prototype is void *calloc(size_t number_of_items, size_t
> size_of_an_item), and scan-build issued a warning that the element size
> was wrong
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/3699/files#r37730074>.
>